SharkCage/CageManager/CageManager.cpp

```diff
@@ -54,55 +76,36 @@ int main()
 	std::wstring result_data;
 	network_manager.Send(sender, CageMessage::RESPONSE_SUCCESS, L"", result_data);
 
-	SecuritySetup security_setup;
-	auto security_attributes = security_setup.GetSecurityAttributes();
-
-	if (!security_attributes.has_value())
-	{
-		std::cout << "Could not get security attributes" << std::endl;
-		return 1;
-	}
 
 	const int work_area_width = 300;
 	std::thread desktop_thread(
 		&CageManager::StartCage,
 		cage_manager,
 		security_attributes.value(),
-		cage_data
+		cage_data,
+		group_name
 	);
 
 	desktop_thread.join();
-
+	tokenLib::deleteLocalGroup(const_cast<wchar_t*>((group_name.c_str())));
 	return 0;
 }
 
-void CageManager::StartCage(SECURITY_ATTRIBUTES security_attributes, const CageData &cage_data)
+void CageManager::StartCage(SECURITY_ATTRIBUTES security_attributes, const CageData &cage_data, const std::wstring &group_name)
 {
 	// name should be unique every time -> create UUID
-	UUID uuid;
-	if (::UuidCreate(&uuid) != RPC_S_OK)
-	{
-		std::cout << "Failed to create UUID" << std::endl;
+	auto uuid_stl_opt = generateUuid();
+	if (!uuid_stl_opt.has_value()) {
 		return;
 	}
-
-	RPC_WSTR uuid_str;
-	if (::UuidToString(&uuid, &uuid_str) != RPC_S_OK)
-	{
-		std::cout << "Failed to convert UUID to rpc string" << std::endl;
+	auto uuid_stl = uuid_stl_opt.value();
+	HANDLE token_handle = nullptr;
+	//TODO: close this handle
```

Comment:
Yes we should close the handle. Is there anything against doing it?

Reply:
Nope, I just forgot 😅